packages/nimble-components/src/rich-text/mention-list-box/index.ts

```diff
@@ -0,0 +1,188 @@
+                    return true;
+                }
+                return false;
+            }
+            case keyArrowDown: {
+                if (!e.shiftKey && this.hasSelectableOptions) {
+                    this.listBox.selectNextOption();
+                    this.scrollOptionIntoView();
+                    return true;
+                }
+                return false;
+            }
+            case keyArrowUp: {
+                if (!e.shiftKey && this.hasSelectableOptions) {
+                    this.listBox.selectPreviousOption();
+                    this.scrollOptionIntoView();
+                    return true;
+                }
+                return false;
+            }
+            default: {
+                return false;
+            }
+        }
+    }
+
+    /**
+     * @public
+     */
+    public clickHandler(e: MouseEvent): boolean {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+        const captured = (e.target as HTMLElement).closest(
+            'option,[role=option]'
+        ) as ListOption | null;
+
+        if (!captured || captured.disabled) {
+            return false;
+        }
+        this.$emit('change', {
+            href: captured.value,
+            displayName: captured.innerText
+        } as MentionDetail);
+        return true;
+    }
+
+    private scrollOptionIntoView(): void {
+        this.listBox?.firstSelectedOption?.scrollIntoView({ block: 'nearest' });
+    }
+
+    private slottedOptionsChanged(
+        _prev: Element[] | undefined,
+        next: Element[]
+    ): void {
+        this.options = next.reduce<ListOption[]>((options, item) => {
+            if (isListboxOption(item)) {
+                options.push(item);
+            }
+            return options;
+        }, []);
+        this.filterUsers();
+    }
+
+    private filterChanged(_prev: string | undefined, next: string): void {
+        this.filter = next;
+        this.filterUsers();
+    }
+
+    private filterUsers(): void {
+        if (this.filter !== undefined) {
+            const normalizedFilter = this.normalizeString(this.filter);
+            this._options.forEach(o => {
+                const normalizedText = this.normalizeString(o.text);
+                const checkFlag = !normalizedText.includes(normalizedFilter);
+                o.disabled = checkFlag;
+                o.hidden = checkFlag;
+            });
+            this.hasSelectableOptions = !this._options.every(
+                list => list.disabled
+            );
+        }
+        void this.selectFirstListOption();
+    }
+
+    private normalizeString(value: string): string {
+        // This implementation is based on the below reference.
+        // https://claritydev.net/blog/diacritic-insensitive-string-comparison-javascript
+        return value
+            .normalize('NFD')
+            .replace(/[\u0300-\u036f]/g, '')
+            .toLowerCase();
+    }
```

Comment:
I'm concerned about this getting out of sync with external usages and potentially even with server-side search in other features.

See discussion thread here: https://dev.azure.com/ni/DevCentral/_git/Skyline/pullRequest/600443#1701308051

Reply:
As we aligned on the above conversation, we will update the `user-service` library to reuse the nimble method for normalization.

Created a task to track this:
https://dev.azure.com/ni/DevCentral/_workitems/edit/2592172